libs/dao/src/main/java/com/akto/dto/ApiCollection.java

```diff
@@ -174,16 +174,19 @@ public void addToConditions(CollectionCondition condition) {
 
         boolean found = false;
 
-        for(CollectionCondition it : conditions){
-            switch(it.getType()){
-                case API_LIST:
-                    Set<ApiInfoKey> tmp = it.returnApis();
-                    tmp.addAll(condition.returnApis());
-                    ((ApiListCondition) it).setApiList(tmp);
-                    found = true;
-                    break;
-                default:
-                    break;
+        for (CollectionCondition it : conditions) {
+            boolean sameType = it.getType() == condition.getType();
```

Comment:
what is this for?

Reply:
When we add/remove an API in a custom collection manually, it should be added to the condition object for the same if the object exists in the conditions list. Since, we would have multiple condition object types, this checks for that.